sdks/python/apache_beam/runners/worker/opcounters.py

```diff
@@ -224,8 +230,25 @@ def _observable_callback_inner(value, is_encoded=False):
 
     return _observable_callback_inner
 
+  def type_check(self, value):
+    # type: (any, bool) -> None
+    for transform_label, type_constraint_tuple in self.output_type_constraints.items():
+      parameter_name, constraint = type_constraint_tuple
+      if constraint is not None:
+        try:
+          _check_instance_type(constraint, value, parameter_name, verbose=True)
+        except TypeCheckError as e:
+          if not transform_label.startswith('ParDo'):
```

Comment:
I don't think we want to mess with this. (Also, if we do, let's push this logic up into the visitor where we decide the label.)

Reply:
Okay, I pushed the logic up. The reason is so we stay consistent with how the [existing error message](https://github.com/apache/beam/blob/3e7371eae49b7aba142c8c0e898a62f583b49276/sdks/python/apache_beam/typehints/typecheck.py#L93) is formatted.

Reply:
Oh... Yeah, changing all the error messages would be a pain, even if they're wrong now. Can you file a JIRA and drop a TODO to fix that then?

Reply:
Sure - who should I assign the TODO to?